presentation/src/main/java/com/martgant/tolkienquizapp/QuestionFragment.kt

```diff
@@ -0,0 +1,76 @@
+package com.martgant.tolkienquizapp
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.StringRes
+import androidx.fragment.app.Fragment
+import com.martgant.tolkienquizapp.databinding.FragmentQuestionBinding
+import kotlin.random.Random
+
+class QuestionFragment : Fragment() {
+
+    private var _binding: FragmentQuestionBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        _binding = FragmentQuestionBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val (questionRes, answer1Res, answer2Res) = getQuestionProps()
+        binding.questionDescription.text = context?.getString(questionRes)
+        binding.answer1.text = context?.getString(answer1Res)
+        binding.answer2.text = context?.getString(answer2Res)
+    }
+
+    private fun getQuestionProps(): QuestionProps {
+        val questionNumber = Random.nextInt(1, 5)
+
+        val questionFormat = null
+        val answerCorrectFormat = null
+        val answerIncorrectFormat = null
+        return QuestionProps(
+            questionRes = resources.getIdentifier(
+                questionFormat.format(questionNumber),
```

Comment:
These 3 lines won't build - `questionFormat` is null in here (also complier doesn't know if it's `String?`, `Int?` etc since no type has been declared). What you should use here is `QUESTION_FORMAT`. 
Similar thing in lines 48 and 53